components/pages/explorer/PageContent.tsx

```diff
@@ -82,8 +84,42 @@ const PageContent = () => {
 	const isFileTableActive = tableType === tableTypes['REPO_TABLE'];
 
 	const switchTable = () => {
+		const { selectedRows, tableData } = tableContext;
 		const nextTableValue = isFileTableActive ? tableTypes['BAM_TABLE'] : tableTypes['REPO_TABLE'];
-		setTableType(nextTableValue);
+
+		if (nextTableValue === tableTypes['BAM_TABLE']) {
+			const oneFileSelected = selectedRows.length === 1;
+
+			if (selectedRows.length === 0) {
+				Router.push({
+					pathname: '/_error',
+				});
+			} else if (!oneFileSelected) {
+				Router.push({
+					pathname: '/_error',
+				});
+			}
```

Comment:
Isn't this condition chain equivalent to the following (simpler) suggestion?
```suggestion
			if (!oneFileSelected) {
				return Router.push({
					pathname: '/_error',
				});
			}
```

...asides from which, is this version of NextJS so outdated that we don't have the `useRouter` hook available?
https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#userouter-hook